renderdoc/driver/gl/gl_pixelhistory.cpp

```diff
@@ -340,14 +679,55 @@ rdcarray<PixelModification> GLReplay::PixelHistory(rdcarray<EventUsage> events,
   }
 
   QueryFailedTests(m_pDriver, resources, modEvents, x, flippedY, history);
-  QueryPostModPixelValues(m_pDriver, resources, modEvents, x, flippedY, history);
+  std::map<uint32_t, uint32_t> eventFragments =
+      QueryNumFragmentsByEvent(m_pDriver, resources, modEvents, x, flippedY);
+
+  // copy history entries to create one history per fragment
+  for(size_t h = 0; h < history.size();)
+  {
+    uint32_t frags = 1;
+    auto it = eventFragments.find(history[h].eventId);
+    if(it != eventFragments.end())
+    {
+      frags = it->second;
+    }
+    for(uint32_t f = 1; f < frags; f++)
```

Comment:
Minor, but we seem to be using i++ in some places and ++i in other places, we should standardize that